src/main/java/com/google/cloud/resourcemanager/LienInfo.java

```diff
@@ -0,0 +1,210 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Lien;
+import com.google.common.base.Function;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager lien metadata object.
+ *
+ * <p>A Lien represents an encumbrance on the actions that can be performed on a resource.
+ */
+public class LienInfo {
+
+  static final Function<Lien, LienInfo> FROM_PROTOBUF_FUNCTION =
+      new Function<Lien, LienInfo>() {
+        @Override
+        public LienInfo apply(Lien lienProtobuf) {
+          return LienInfo.fromProtobuf(lienProtobuf);
+        }
+      };
+  static final Function<LienInfo, Lien> TO_PROTOBUF_FUNCTION =
+      new Function<LienInfo, Lien>() {
+        @Override
+        public Lien apply(LienInfo lienInfo) {
+          return lienInfo.toProtobuf();
+        }
+      };
+
+  private String createTime;
+  private String name;
+  private String origin;
+  private String parent;
+  private String reason;
+  private List<String> restrictions;
+
+  /** Builder for {@code LienInfo}. */
+  static class Builder {
+
+    private String createTime;
+    private String name;
+    private String origin;
+    private String parent;
+    private String reason;
+    private List<String> restrictions;
+
+    Builder(String parent) {
+      this.parent = parent;
+    }
+
+    Builder(LienInfo info) {
+      this.createTime = info.createTime;
+      this.name = info.name;
+      this.origin = info.origin;
+      this.parent = info.parent;
+      this.reason = info.reason;
+      this.restrictions = info.restrictions;
+    }
+
+    public Builder setCreateTime(String createTime) {
+      this.createTime = createTime;
+      return this;
+    }
+
+    public Builder setName(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder setOrigin(String origin) {
+      this.origin = origin;
+      return this;
+    }
+
+    public Builder setReason(String reason) {
+      this.reason = reason;
+      return this;
+    }
+
+    public Builder setRestrictions(List<String> restrictions) {
+      this.restrictions = restrictions;
+      return this;
+    }
+
+    public Builder setParent(String parent) {
+      this.parent = checkNotNull(parent);
+      return this;
+    }
+
+    public LienInfo build() {
+      return new LienInfo(this);
+    }
+  }
+
+  LienInfo(Builder builder) {
+    this.createTime = builder.createTime;
+    this.name = builder.name;
+    this.origin = builder.origin;
+    this.parent = builder.parent;
+    this.reason = builder.reason;
+    this.restrictions = builder.restrictions;
```

Comment:
This exposes potentially mutable internal state so making a copy is a good idea.